src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpSystemProxy.cs

```diff
@@ -131,8 +156,59 @@ public Uri GetProxy(Uri uri)
         {
             if (_proxyHelper.ManualSettingsOnly)
             {
+                if ( _bypassLocal)
+                {
+                    IPAddress address = null;
+
+                    if (uri.IsLoopback)
+                    {
+                        // This is optimization for loopback addresses.
+                        // Unfortunately this does not work for all local addresses.
+                        return null;
+                    }
+                    if (uri.Host[0] == '[' || Char.IsNumber(uri.Host[0]))
+                    {
+                        // RFC1123 allows labels to start with number.
+                        // Leading number may or may not be IP address.
+                        // IPv6 [::1] notation. '[' is not valid character in names.
+                        IPAddress.TryParse(uri.Host, out address);
+                    }
+                    if (address != null)
```

Comment:
address will only be non-null here if TryParse succeeded, so wouldn't it be better for this to be an if around the TryParse in the above block instead?